### PR TITLE
[CSM-279] Don't remove spent wallet addresses

### PR DESCRIPTION
**Pull request by Martoon-00**
This also implements `isUsed` wallet address field properly

### CODE REVIEW

**Comment by Martoon-00** [comment]
To be tested once #902 is merged (it fixes genesis address, which is important here)

File: src/Pos/Wallet/Web/Server/Methods.hs [hunk 1]
```diff
@@ -388,7 +388,12 @@ getWAddressBalance addr =
 getWAddress :: WalletWebMode m => CWAddressMeta -> m CAddress
 getWAddress cAddr@CWAddressMeta{..} = do
     balance <- getWAddressBalance cAddr
-    return $ CAddress cwamId (mkCCoin balance) (balance > minBound)
+    -- TODO [CSM-288]: this is super bad, extract required logic from 'getHistory'.
+    -- TODO [CSM-288]: this is very inefficient as well.
+    (ctxs, _) <- getHistory (walletAddrMetaToAccount cAddr) Nothing (Just 999999)
```
Comment (pva701) [inline src/Pos/Wallet/Web/Server/Methods.hs]: lol, 999999

**Comment by pva701** [comment]
@Martoon-00 why the implementation of tracking stayed not changed?

**Comment by pva701** [comment]
@Martoon-00 maybe should we wait for a reply from @darko-mijic in the YT (CSM-279)?

**Comment by Martoon-00** [comment]
> why the implementation of tracking stayed not changed?

What do you mean? I removed removals of `address`es

**Comment by pva701** [comment]
>What do you mean? I removed removals of addresses
ah, ok

**Comment by Martoon-00** [comment]
WIP: to be tested once possible

**Comment by akegalj** [comment]
@Martoon-00 @pva701 does this https://issues.serokell.io/issue/CSM-279#comment=96-4367 resolves our questions ?

**Comment by Martoon-00** [comment]
I can complete and merge it now. If it will appear to be inconvenient or bad, I propose to deal with it just then

**Comment by akegalj** [comment]
@Martoon-00 ok, lets do it that way

**Comment by akegalj** [comment]
@Martoon-00 we should have 0.5-rc for a base branch. What I suggest is:

    create branch X out of 0.5-rc
    cherry pick this commit to X
    make PR

I am not sure how changing base branch is working in github but it pulls a big diff in this branch

**Comment by Martoon-00** [comment]
Close in favour of #929 (to release branch)

**Pull request closed without merging**